File: sdk/java/src/main/java/vn/hitechcloud/sdk/resources/Collocation.java
package vn.hitechcloud.sdk.resources;

import java.util.Map;
import vn.hitechcloud.sdk.HttpClient;

public class Collocation extends BaseResource {

    public Collocation(HttpClient http) {
        super(http);
    }

    /** Get PDU info */
    public Map<String, Object> pdu(int id) throws Exception {
        return http.get(String.format("/api/service/%id/pdu", id));
    }

    /** Get PDU port */
    public Map<String, Object> pduPort(int id, int port) throws Exception {
        return http.get(String.format("/api/service/%id/pdu/%port", id, port));
    }

    /** Toggle PDU port */
    public Map<String, Object> togglePduPort(int id, int port, Map<String, Object> data) throws Exception {
        return http.post(String.format("/api/service/%id/pdu/%port", id, port), data);
    }

}

File: sdk/nodejs/src/resources/collocation.js
"use strict";

const { BaseResource } = require("./base");

class CollocationResource extends BaseResource {
  async listPdu(serviceId) {
    return this._http.get("/api/service/{id}/pdu");
  }

  async getPduPort(serviceId, port) {
    return this._http.get(`/api/service/{id}/pdu/${port}`);
  }

  async setPduPort(serviceId, port, data) {
    return this._http.post(`/api/service/{id}/pdu/${port}`, data);
  }

}

module.exports = { CollocationResource };
